src/context/historyContext.js

```diff
@@ -0,0 +1,76 @@
+import axios from "axios"
+import React, { createContext, useContext, useReducer } from "react";
+import { historyVideoReducer } from "../reducer";
+
+const defaultValue = {
+    loader: false,
+    historyVideoInfo: [],
+    error: false,
+    errorMsg: null
+};
+
+const HistoryVideoContext = createContext(defaultValue);
+
+const HistoryVideoProvider = ({ children }) => {
+
+    const [ state, dispatch  ] = useReducer(historyVideoReducer, defaultValue);
+
+    const getHistoryVideo = async () => {
+        try {
+            dispatch({ type: "HISTORY_VIDEO_REQUEST" })
+            const token = localStorage.getItem("token");
```

Comment:
suggestion: call to localStorage is synchronous so it is better to extract this data once and reuse it.

Reply:
Sure